### ISSUE TITLE
App crashes on startup in android when Code shrinker is set to r8

### ISSUE THREAD

**Issue by salarcode** [dotnet/maui#8454]
### Description

Application fails in startup when I set the set the `code shrinker` to `r8`

To see  the exception run the app in Debug mode in Android Emulator

### Steps to Reproduce

1. Create a File > New .NET MAUI App
2. Set code shrinker for Android to `r8` either from project `.csproj` file 
```
<PropertyGroup Condition="'$(TargetFramework)|$(Platform)'=='net6.0-android|AnyCPU'">
	<AndroidLinkTool>r8</AndroidLinkTool>
</PropertyGroup> 
```
or from project properties set `Code shrinker` to `r8`
![image](https://user-images.githubusercontent.com/1272095/176778349-7d30b18b-e2ff-4c3c-87ab-fa46b418d041.png)

4. Try to run in Android Emulator in debug mode to the see the exception

Expected outcome: Working application
Actual outcome: Application crashes on startup

### Version with bug

6.0.400 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android

### Did you find any workaround?

Don't set `Code shrinker`

### Relevant log output

```shell
[MonoDroid] UNHANDLED EXCEPTION:
[MonoDroid] Java.Lang.NullPointerException: Attempt to invoke virtual method 'void androidx.fragment.app.FragmentManagerViewModel.setIsStateSaved(boolean)' on a null object reference
[MonoDroid] 
[MonoDroid]   --- End of managed Java.Lang.NullPointerException stack trace ---
[MonoDroid] java.lang.NullPointerException: Attempt to invoke virtual method 'void androidx.fragment.app.FragmentManagerViewModel.setIsStateSaved(boolean)' on a null object reference
[MonoDroid] 	at androidx.fragment.app.FragmentManager.dispatchActivityCreated(FragmentManager.java:2783)
[MonoDroid] 	at androidx.fragment.app.FragmentController.dispatchActivityCreated(FragmentController.java:262)
[MonoDroid] 	at androidx.fragment.app.FragmentActivity.onStart(FragmentActivity.java:478)
[MonoDroid] 	at androidx.appcompat.app.AppCompatActivity.onStart(AppCompatActivity.java:246)
[MonoDroid] 	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1455)
[MonoDroid] 	at android.app.Activity.performStart(Activity.java:8076)
[MonoDroid] 	at android.app.ActivityThread.handleStartActivity(ActivityThread.java:3660)
[MonoDroid] 	at android.app.servertransaction.TransactionExecutor.performLifecycleSequence(TransactionExecutor.java:221)
[MonoDroid] 	at android.app.servertransaction.TransactionExecutor.cycleToPath(TransactionExecutor.java:201)
[MonoDroid] 	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:173)
[MonoDroid] 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
[MonoDroid] 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2210)
[MonoDroid] 	at android.os.Handler.dispatchMessage(Handler.java:106)
[MonoDroid] 	at android.os.Looper.loopOnce(Looper.java:201)
[MonoDroid] 	at android.os.Looper.loop(Looper.java:288)
[MonoDroid] 	at android.app.ActivityThread.main(ActivityThread.java:7839)
[MonoDroid] 	at java.lang.reflect.Method.invoke(Native Method)
[MonoDroid] 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
[MonoDroid] 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1003)
```

**Comment by salarcode** [comment]
Some additional unexpected behaviour is when I set Code shrinker from project properties it sets the android r8 for other platforms as well 🤔

```
	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net6.0-android|AnyCPU'">
	  <AndroidLinkTool>r8</AndroidLinkTool>
	</PropertyGroup>

	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net6.0-android|AnyCPU'">
	  <AndroidLinkTool>r8</AndroidLinkTool>
	</PropertyGroup>

	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net6.0-ios|AnyCPU'">
	  <AndroidLinkTool>r8</AndroidLinkTool>
	</PropertyGroup>

	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net6.0-ios|AnyCPU'">
	  <AndroidLinkTool>r8</AndroidLinkTool>
	</PropertyGroup>

	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Debug|net6.0-windows10.0.19041.0|AnyCPU'">
	  <AndroidLinkTool>r8</AndroidLinkTool>
	</PropertyGroup>

	<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net6.0-windows10.0.19041.0|AnyCPU'">
	  <AndroidLinkTool>r8</AndroidLinkTool>
	</PropertyGroup>
```

**Comment by jonathanpeppers** [comment]
@salarcode it's expected that you'll have to write a few proguard rules a the moment. Did you try adding some for your app?

https://docs.microsoft.com/xamarin/android/deploy-test/release-prep/proguard#customizing-proguard

`r8` uses the same rule format as `proguard` did.

We're trying to make `r8` work out of the box, but that probably won't come until .NET 7.

I'd file an issue via `Help > Send Feedback > Report a Problem` in VS about what it's doing to your project file. Link to it here, and we can upvote it. Thanks!

**Comment by salarcode** [comment]
Following the link above it seems I have to use Release mode only.
So using Release mode both `r8` and `proguard` work fine in Android Emulator.
and now i'm limiting the shrinker to Release mode only
```
<PropertyGroup Condition="'$(Configuration)|$(TargetFramework)|$(Platform)'=='Release|net6.0-android|AnyCPU'">
	<AndroidLinkTool>proguard</AndroidLinkTool>
</PropertyGroup>
```

I think question comes down to VS why it is generating the configuration for Debug and also other platforms!

I've filed a report for VS issue
https://developercommunity.visualstudio.com/t/Setting-Android-Code-Shrinker-should-not/10086369?entry=problem 

Thanks

**Comment by salarcode** [comment]
I was trying to use r8, it sound ok to be a checkbox.
The main problem here was that VS is generating incorrect config in .csproj which I have [created an issue](https://developercommunity.visualstudio.com/t/Setting-Android-Code-Shrinker-should-not/10086369?entry=problem) about it.
Thanks

**Comment by adamint** [comment]
Hi @salarcode, that's not a project system bug. While yes, all configs are being generated for that property, that's because _there is no dimension that the value is conditioned on_, which you can do by clicking the gear icon on the property and selecting "Configuration," which lets you put a different release/debug value. The bug you should file should be against @japarson's team of not allowing the release config to be selectable/set/using a property interceptor to prevent value being set, whichever they think is most appropriate.

**Comment by Eilon** [comment]
@jonathanpeppers - any idea where this issue should go to?

**Comment by japarson** [comment]
@Eilon My team can handle this via the feedback ticket that was filed.

**Comment by japarson** [comment]
This issue will be fixed in VS 17.6 Preview 2.
